src/library/scala/collection/Parallelizable.scala

```diff
@@ -43,6 +46,31 @@ trait Parallelizable[+A, +ParRepr <: Parallel] extends Any {
     cb.result()
   }
 
+  /** Returns a parallel implementation of this collection using the given thread pool.
+   *
+   * @param pool the thread pool to use for supporting parallelism
+   * @return a parallel implementation of this collection using `pool` for its threading
+   */
+  final def parWith(pool: ForkJoinPool): ParRepr = parWith(new ForkJoinTaskSupport(pool))
+
+  /** Returns a parallel implementation of this collection using the given task support.
+   *
+   * @param ts the task support to use for supporting parallelism
+   * @return a parallel implementation of this collection using `ts` for parallelism
+   */
+  final def parWith(ts: TaskSupport): ParRepr = {
+    val pseq = par
+    setTaskSupport(pseq, ts)
+    pseq
+  }
+
+  /** Returns a parallel implementation of this collection using the given number of threads.
+   *
+   * @param pool the number of threads to use for supporting parallelism
+   * @return a parallel implementation of this collection using `n` threads
+   */
+  final def parWith(n: Int): ParRepr = parWith(new ForkJoinPool(n))
```

Comment:
Although I admit an advantage is without the `Int` overloading we could probably just overload `par`